File: app/src/main/java/br/com/faru/a99challenge/model/Person.java
package br.com.faru.a99challenge.model;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import java.util.Date;

/**
 * Created by faru on 04/04/17.
 */

public class Person implements Parcelable, Comparable<Person> {

    private String id;
    private String name;
    private String image;
    private Date birthday;
    private String bio;

    protected Person(Parcel in) {
        id = in.readString();
        name = in.readString();
        image = in.readString();
        bio = in.readString();
    }

    public static final Creator<Person> CREATOR = new Creator<Person>() {
        @Override
        public Person createFromParcel(Parcel in) {
            return new Person(in);
        }

        @Override
        public Person[] newArray(int size) {
            return new Person[size];
        }
    };

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Date getBirthday() {
        return birthday;
    }

    public void setBirthday(Date birthday) {
        this.birthday = birthday;
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(id);
        dest.writeString(name);
        dest.writeString(image);
        dest.writeString(bio);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Person person = (Person) o;

        return id != null ? id.equals(person.id) : person.id == null;
    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }

    @Override
    public int compareTo(@NonNull Person person) {
        return this.getName().compareTo(person.getName());
    }
}

File: app/src/main/java/br/com/faru/a99challenge/model/Person.kt
package br.com.faru.a99challenge.model

import android.os.Parcel
import android.os.Parcelable

import java.util.Date

data class Person(var id: String? = null,
                  var name: String? = null,
                  var image: String? = null,
                  var birthday: Date? = null,
                  var bio: String? = null) : Parcelable, Comparable<Person> {

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readSerializable() as Date?,
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeString(image)
        parcel.writeSerializable(birthday)
        parcel.writeString(bio)
    }

    override fun describeContents() = 0

    override fun compareTo(other: Person) = other.name?.let { this.name?.compareTo(it) } ?: 0

    companion object CREATOR : Parcelable.Creator<Person> {
        override fun createFromParcel(parcel: Parcel): Person = Person(parcel)
        override fun newArray(size: Int): Array<Person?> = arrayOfNulls(size)
    }
}
